perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -234,28 +243,136 @@ class GceNetwork(network.BaseNetwork):
   def __init__(self, network_spec):
     super(GceNetwork, self).__init__(network_spec)
     self.project = network_spec.project
-    name = FLAGS.gce_network_name or 'pkb-network-%s' % FLAGS.run_uri
-    mode = 'auto' if FLAGS.gce_subnet_region is None else 'custom'
+
+    #  Figuring out the type of network here.
+    #  Precedence: User Managed > CIDR (multi) > gce_subnet flag (single) > Auto (default)
+    self.net_type = 'default'  # Default network 'auto' mode. Single VPC with subnets in all regions
+    self.cidr = NETWORK_RANGE
+    if FLAGS.gce_subnet_region:  # Overrides auto network when set. Creates one VPC network in single region.
+      self.net_type = 'single'
+      self.cidr = FLAGS.gce_subnet_addr
+    if network_spec.cidr:  # Creates multiple VPC networks when set.
+      self.net_type = 'multi'
+      self.cidr = network_spec.cidr
+
+    name = self._MakeGceNetworkName()
+
+    subnet_region = FLAGS.gce_subnet_region if not network_spec.cidr else util.GetRegionFromZone(network_spec.zone)
+    mode = 'auto' if subnet_region is None else 'custom'
     self.network_resource = GceNetworkResource(name, mode, self.project)
-    if FLAGS.gce_subnet_region is None:
+    if subnet_region is None:
       self.subnet_resource = None
     else:
-      self.subnet_resource = GceSubnetResource(name, name,
-                                               FLAGS.gce_subnet_region,
-                                               FLAGS.gce_subnet_addr,
-                                               self.project)
-    firewall_name = 'default-internal-%s' % FLAGS.run_uri
+      self.subnet_resource = GceSubnetResource(name, name, subnet_region, self.cidr, self.project)
+
+    # Stage FW rules.
+    self.all_nets = self._GetNetworksFromSpec(network_spec)  # Holds the different networks in this run.
+    self.external_nets_rules = {}  # Holds FW rules for any external subnets.
+
+    #  Set the default rule to allow all traffic within this network's subnet.
+    firewall_name = self._MakeGceFWRuleName()
     self.default_firewall_rule = GceFirewallRule(
-        firewall_name, self.project, ALLOW_ALL, name, NETWORK_RANGE)
+        firewall_name, self.project, ALLOW_ALL, name, self.cidr)
+
+    # Set external rules to allow traffic from other VPCs/subnets in this benchmark.
+    for ext_net in self.all_nets:
+      if ext_net == self.cidr:
+        continue  # We've already added our own network to the default rule.
+      rule_name = self._MakeGceFWRuleName(dst_cidr=ext_net)
+      self.external_nets_rules[rule_name] = GceFirewallRule(rule_name, self.project, ALLOW_ALL, name, ext_net)
+
+
+  def _GetNetworksFromSpec(self, network_spec):
+    """
+    Returns a list of distinct CIDR networks defined in this benchmark's vm_groups.
```

Comment:
please format per http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings
i.e. single line for 
`"""Returns ... vm_groups.`

Reply:
Fixed.